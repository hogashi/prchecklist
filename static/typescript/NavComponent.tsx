import * as React from 'react';
import * as API from './api';

interface NavProps {
  logo?: JSX.Element;
  stages?: JSX.Element;
  me?: API.GitHubUser;
}

interface NavState {
  version: string;
}

export class NavComponent extends React.Component<NavProps, NavState> {
  public componentDidMount() {
    this.setState({
      version: document.querySelector('meta[name="prchecklist version"]').getAttribute('content'),
    });
  }

  public render() {
    return <nav>
      <div className="logo"><strong>{this.props.logo || `prchecklist ${this.state.version}`}</strong></div>
      <div className="stages">{this.props.stages}</div>
      {
        this.props.me
          ? <div className="user-signedin">{this.props.me.Login}</div>
          : <a className="user-signedin" href={`/auth?return_to=${encodeURIComponent(location.pathname)}`}>Login</a>
      }
    </nav>;
  }
}
