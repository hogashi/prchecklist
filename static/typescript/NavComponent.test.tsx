import * as React from "react";
import * as renderer from "react-test-renderer";

import { NavComponent } from "./NavComponent";

test("", async () => {
  const component = renderer.create(<NavComponent />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
