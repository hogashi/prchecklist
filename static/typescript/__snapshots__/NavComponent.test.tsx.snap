// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<nav>
  <div
    className="logo"
  >
    <strong>
      prchecklist undefined
    </strong>
  </div>
  <div
    className="stages"
  />
  <a
    className="user-signedin"
    href="/auth?return_to=%2F"
  >
    Login
  </a>
</nav>
`;
